### ISSUE TITLE
Is there a read_state implementation for system calls?

### ISSUE THREAD

**Issue by bodily11** [rocklabs-io/ic-py#40]
Making query and update calls are great, but sometimes I want to make system calls, which requires read_state syntax as can be seen here: https://smartcontracts.org/docs/interface-spec/index.html#http-read-state

Is this interface implemented in the Python agent? I know the JavaScript agent tests supported some read_state calls, but the documentation around it still wasn't very good.

Thanks!

**Comment by ccyanxyz** [comment]
https://github.com/rocklabs-io/ic-py/blob/main/ic/agent.py#L107

**Comment by bodily11** [comment]
Yeah, beautiful. That is the one that I found as well. Although "time" is the only path that I could get to work. I kept getting CBOR decode issues if I tried any other paths. Are other paths other than "time" supported? Like canister_ranges? Or subnet information?

**Comment by Myse1f** [comment]
Canister info and subnet informatoin are supported. Text in paths must be blob and principal in the paths must be bytes-form,  e.g

```python
cert = ag.read_state_raw("gvbup-jyaaa-aaaah-qcdwa-cai", [["subnet".encode(), Principal.from_str("w4asl-4nmyj-qnr7c-6cqq4-tkwmt-o26di-iupkq-vx4kt-asbrx-jzuxh-4ae").bytes, "canister_ranges".encode()]])
ret = lookup(["subnet".encode(), Principal.from_str("w4asl-4nmyj-qnr7c-6cqq4-tkwmt-o26di-iupkq-vx4kt-asbrx-jzuxh-4ae").bytes, "canister_ranges".encode()], cert)
print(cbor2.loads(ret))
```

Actually, `read_state_raw` is too low level. Maybe we can absract it into easy-to-use function later.

**Comment by Myse1f** [comment]
#41 
You can look into the code in `test_readstate` and try ur own code to check whether get correct result.
